### DISCUSSION TITLE
How can I customize fonts?

### DISCUSSION THREAD

**Discussion by SHDocter** [zauberzeug/nicegui#2208] [Q&A] [answered]
### Question

I want to customize the font, but I don't know how to do it...

It's best to use badge. If it doesn't work, the label can also be used. Thanks!

**Comment by retsyo** [comment]
```python
from nicegui import ui

ui.label('hello').style('font-family: Symbol; font-size: 40px; font-weight: bold;')

with ui.button('world').style('font-family: Symbol; font-size: 40px; font-weight: bold;'):
    ui.badge("0", color='red').props('floating').style('font-family: Symbol; font-size: 1em; font-weight: bold;')

ui.run()
```

**Reply by SHDocter** [reply]
But where should I put the font file? Is it placed in the root directory of the program?

**Comment by retsyo** [comment, marked as answer]
# simple answer
after a quick search, I coined the code:
```python
from nicegui import ui, app

app.add_static_files('/path_in_webserver', r'C:\Windows\Fonts')

ui.add_head_html(r'''
<style>
@font-face{
    font-family: "my font";
    src: url('/path_in_webserver/Symbol.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
}
</style>
''')

ui.label('hello').style('font-family: "my font"; font-size: 40px; font-weight: bold;')

with ui.button('world').style('font-family: "my font"; font-size: 40px; font-weight: bold;'):
    ui.badge("0", color='red').props('floating').style('font-family: "my font"; font-size: 1em; font-weight: bold;')

ui.run()
```
 in which you can read the keys are
- `app.add_static_files`, so you can put your font files in any folds
- `ui.add_head_html`

I test the code in LAN, it renders as expected on client where there is no local `Symbol.ttf` copy

# not that simple answer
but the quick search also tells me a jargon `Cross-Origin Resource Sharing (CORS) `, which is beyond my time, my experience and testing environment( currently I only use my code in LAN. In other words, I have not deployed my code on any sever yet). So you have to do some investigation by yourself. 

Some useful, at least I think so, links in the order of "usefulness"
- https://fastapi.tiangolo.com/tutorial/cors/
- https://stackoverflow.com/questions/65635346/how-can-i-enable-cors-in-fastapi
- https://www.hirehop.com/blog/cross-domain-fonts-cors-font-face-issue/
- https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

# good luck
And feedback is welcome.

**Reply by SHDocter** [reply]
I don't need to access it on the internet, so I don't have to worry about CORS. Your answer is very useful, thank you very much for your help!

**Reply by falkoschindler** [reply]
I think CORS would only be an issue if you would want to load the font from another host. But if you serve the font file yourself using `ui.add_static_files`, the font is loaded from the same host like the rest of the page, so CORS does not apply.
